src/Compression/CompressionCodecDeflate.h

```diff
@@ -0,0 +1,245 @@
+        return (static_cast<size_t>((tsc * 44485709377909ULL) >> 4)) % pool_size;
+    }
+
+    int32_t ALWAYS_INLINE get_job_size_helper()
+    {
+        static uint32_t size = 0;
+        if (size == 0)
+        {
+            const auto status = qpl_get_job_size(PATH, &size);
+            if (status != QPL_STS_OK)
+            {
+                return -1;
+            }
+        }
+        return size;
+    }
+
+    int32_t ALWAYS_INLINE init_job_helper(qpl_job * qpl_job_ptr)
+    {
+        if (qpl_job_ptr == nullptr)
+        {
+            return -1;
+        }
+        auto status = qpl_init_job(PATH, qpl_job_ptr);
+        if (status != QPL_STS_OK)
+        {
+            return -1;
+        }
+        return 0;
+    }
+
+    int32_t ALWAYS_INLINE initJobPool()
+    {
+        static bool initialized = false;
+
+        if (initialized == false)
+        {
+            const int32_t size = get_job_size_helper();
+            if (size < 0)
+                return -1;
+            for (int i = 0; i < jobPoolSize; ++i)
+            {
+                jobPool[i] = nullptr;
+                qpl_job * qpl_job_ptr = reinterpret_cast<qpl_job *>(new uint8_t[size]);
+                if (init_job_helper(qpl_job_ptr) < 0)
+                    return -1;
+                jobPool[i] = qpl_job_ptr;
+                jobLock[i].store(false);
+            }
+            initialized = true;
+        }
+        return 0;
+    }
+
+    bool ALWAYS_INLINE tryLockJob(size_t index)
+    {
+        bool expected = false;
+        return jobLock[index].compare_exchange_strong(expected, true);
+    }
+
+    void ALWAYS_INLINE destroyJobPool()
+    {
+        const uint32_t size = get_job_size_helper();
+        for (uint32_t i = 0; i < jobPoolSize && size > 0; ++i)
+        {
+            while (tryLockJob(i) == false)
+            {
+            }
+            if (jobPool[i])
+            {
+                qpl_fini_job(jobPool[i]);
+                delete[] jobPool[i];
+            }
+            jobPool[i] = nullptr;
+            jobLock[i].store(false);
+        }
+    }
+
+    struct ReleaseJobObjectGuard
+    {
+        uint32_t index;
+        ReleaseJobObjectGuard() = delete;
+
+    public:
+        ALWAYS_INLINE ReleaseJobObjectGuard(const uint32_t i) : index(i)
+        {
+        }
+        ALWAYS_INLINE ~ReleaseJobObjectGuard()
+        {
+            jobLock[index].store(false);
+        }
+    };
+    Poco::Logger * log;
+};
+class SoftwareCodecDeflate
+{
+public:
+    SoftwareCodecDeflate();
+    ~SoftwareCodecDeflate();
+    uint32_t doCompressData(const char * source, uint32_t source_size, char * dest, uint32_t dest_size);
+    void doDecompressData(const char * source, uint32_t source_size, char * dest, uint32_t uncompressed_size);
+
+private:
+    qpl_job * jobSWPtr; //Software Job Codec Ptr
+    std::unique_ptr<uint8_t[]> jobSWbuffer;
+    qpl_job * getJobCodecPtr();
```

Comment:
Because software job can be initialized only on demand, which mean it don't need do initialize on construction. So I think here just define getJobCodecPtr is enough which will initialize sw job and return the value only in case being called.